### PR TITLE
Add mortar & howitzer shots to end game stats

### PR DESCRIPTION
**Pull request by lbnesquik**
## About The Pull Request
Mortar & Howitzers go boom. It would be nice to know if there has been like, 5 shots or 50 during a particular round.

## Why It's Good For The Game
Stats good. Not a huge performance impact given we track bullets fired.

## Changelog
:cl:
add: Added mortar & howitzers shots to end round.
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/mortar.dm [hunk 1]
```diff
@@ -210,6 +210,14 @@
 		flick(icon_state + "_fire", src)
 		mortar_shell.forceMove(src)
 
+		if(istype(I, /obj/item/mortal_shell/howitzer))
+			GLOB.round_statistics.howitzer_shells_fired++
+			SSblackbox.record_feedback("tally", "round_statistics", 1, "howitzer_shells_fired")		
+
+		else if(istype(I, /obj/item/mortal_shell))
```
Comment (QualityVan) [inline code/game/objects/machinery/mortar.dm:217]: This block already knows that I is a shell from 157, you can just have this line as an else.

File: code/game/objects/machinery/mortar.dm [hunk 2]
```diff
@@ -210,6 +210,13 @@
 		flick(icon_state + "_fire", src)
 		mortar_shell.forceMove(src)
 
+		if(istype(src, /obj/machinery/deployable/mortar/howitzer))
```
Comment (TiviPlus) [inline code/game/objects/machinery/mortar.dm:213]: Istype src is bad

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/mortar.dm (hunk 2 above)
Comment (BraveMole) [reply code/game/objects/machinery/mortar.dm:213]: Two solutions : 
Either you override the proc for howitzer

Else, you add a var on the projectile or mortar to specify on which tally it should be added

File: code/game/objects/machinery/mortar.dm [hunk 3]
```diff
@@ -39,6 +39,8 @@
 		/obj/item/mortal_shell/plasmaloss,
 	)
 
+	var/tally_type = 1	//Used for round stats. 1 is mortar, 2 is howitzer
```
Comment (BraveMole) [inline code/game/objects/machinery/mortar.dm:42]: you need autodoc, and you need to use define
///Used for round stats
var/tally_type = MORTAR_ROUND_TALLY

File: code/game/objects/machinery/mortar.dm [hunk 4]
```diff
@@ -210,6 +212,13 @@
 		flick(icon_state + "_fire", src)
 		mortar_shell.forceMove(src)
 
+		if(tally_type == 1)
+			GLOB.round_statistics.howitzer_shells_fired++
+			SSblackbox.record_feedback("tally", "round_statistics", 1, "howitzer_shells_fired")		
+		if(tally_type == 2)
```
Comment (BraveMole) [inline code/game/objects/machinery/mortar.dm:218]: should be elseif or even else

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/mortar.dm [hunk 5]
```diff
@@ -1,6 +1,9 @@
 //The Marine mortar, the T-50S Mortar
 //Works like a contemporary crew weapon mortar
 
+#define MORTAR 1
```
Comment (BraveMole) [inline code/game/objects/machinery/mortar.dm:4]: In general, you want your vars /defines to be as verbose as possible.

Here, TALLY_MORTAR would be a better name

File: code/game/objects/machinery/mortar.dm [hunk 6]
```diff
@@ -39,6 +42,9 @@
 		/obj/item/mortal_shell/plasmaloss,
 	)
 
+	///Used for round stats
+	var/tally_type = MORTAR
+
 
 	use_power = NO_POWER_USE
```
Comment (BraveMole) [inline code/game/objects/machinery/mortar.dm:49]:
```suggestion
	use_power = NO_POWER_USE
	///Used for round stats
	var/tally_type = MORTAR
```

New vars after existing ones

**Review by BraveMole — CHANGES_REQUESTED** [review]
You also need to undefine the defines at the end of the file, since they are only used there

**Review by BraveMole — APPROVED** [review]

**Pull request merged**